src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -697,6 +697,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\Emit\TypeBuilder.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\Emit\TypeBuilderInstantiation.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\Emit\TypeNameBuilder.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\Emit\TypeBuilder.Static.cs" Condition="'$(FeatureNativeAot)' != 'true'" />
```

Comment:
Is there anything preventing us from using this implementation for native AOT as well? (Together with MethodOnTypeBuilderInstantiation.cs and FieldOnTypeBuilderInstantiation.cs.)

Reply:
Actually it is references `TypeBuilderInstantiation` which references `MethodOnTypeBuilderInstantiation`, `FieldOnTypeBuilderInstantiation` and `ConstructorOnTypeBuilderInstantiation` that are excluded from AOT build, probably because they references `RuntimeMethodInfo`, `RuntimeFieldInfo` and `RuntimeConstructorInfo` that are not exist in AOT corelib

Reply:
As far as I can tell, the Runtime...Infos are only referenced by asserts. We should delete the problematic asserts. (I do not think that these asserts are valid - it should be possible to write a test that makes them to fail.)

Reply:
Done